test/integration/component/test_volumes.py

```diff
@@ -603,7 +603,144 @@ def test_01_volume_attach_detach(self):
                 "Check the state of VM"
             )
         except Exception as e:
-            self.fail("Exception occuered: %s" % e)
+            self.fail("Exception occurred: %s" % e)
+        return
+    
+    @attr(tags=["advanced", "advancedns"], required_hardware="false")
+    def test_02_root_volume_attach_detach(self):
+        """Test Root Volume attach/detach to VM
+        """
+
+        # Validate the following
+        # 1. Deploy a VM
+        # 2. Verify that we are testing a supported hypervisor
+        # 3. Check for root volume
+        # 4. Stop VM
+        # 5. Detach root volume
+        # 6. Verify root volume detached
+        # 7. Attach root volume
+        # 8. Start VM
+        
+        # Verify we are using a supported hypervisor
+        if (self.hypervisor.lower() == 'vmware'
+                or self.hypervisor.lower() == 'kvm'
+                or self.hypervisor.lower() == 'simulator'
+                or self.hypervisor.lower() == 'xenserver'):
+        
+            try:
+                # Check for root volume
+                root_volume_response = Volume.list(
+                    self.apiclient,
+                    virtualmachineid=self.virtual_machine.id,
+                    type='ROOT',
+                    listall=True
+                )
+                self.assertNotEqual(
```

Comment:
@nitt10prashant - Thanks for the feedback.  I was merely following an established pattern that I observed earlier in the file.  Please look at lines 499 - 510 as an example.  I have put the code below for your convenience.  Should that code be refactored as well? I just want to understand if my situation that I'm testing is different than the code I'm referencing below.  In the end, I'm happy to make the change.  Thanks again for the feedback.

`
                # Check List Volume response for newly created volume
                list_volume_response = Volume.list(
                    self.apiclient,
                    id=volume.id
                )
                self.assertNotEqual(
                    list_volume_response,
                    None,
                    "Check if volume exists in ListVolumes")
                self.assertEqual(
                    isinstance(list_volume_response, list),
                    True,
                    "Check list volumes response for valid list")`